CommonTools/PileupAlgos/plugins/PuppiPhoton.cc

```diff
@@ -123,6 +124,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
         if(pupCol->refAt(iPF).key() != *itPho) continue;
         pWeight = weight_;
         if(!useValueMap_ && itPF->pt() != 0) pWeight = pWeight*(phoCands[iPho]->pt()/itPF->pt());
+	if(!useValueMap_ && itPF->pt() == 0) pVec.SetPxPyPzE(phoCands[iPho]->px()*pWeight,phoCands[iPho]->py()*pWeight,phoCands[iPho]->pz()*pWeight,phoCands[iPho]->energy()*pWeight);
```

Comment:
Maria's plots are based on the new tune. We know the performance is fine.
In fact, we also knowt hat commenting this line makes things much worse.
The issue does not appear with Puppi, but with whatever the MET code is
doing after Puppi.

On Wed, Sep 7, 2016 at 3:54 AM, Slava Krutelyov <notifications@github.com>
wrote:

> In CommonTools/PileupAlgos/plugins/PuppiPhoton.cc
> <https://github.com/cms-sw/cmssw/pull/15562#discussion_r77749405>:
>
> > @@ -123,6 +124,7 @@ void PuppiPhoton::produce(edm::Event& iEvent, const edm::EventSetup& iSetup) {
> >          if(pupCol->refAt(iPF).key() != *itPho) continue;
> >          pWeight = weight_;
> >          if(!useValueMap_ && itPF->pt() != 0) pWeight = pWeight*(phoCands[iPho]->pt()/itPF->pt());
> > +	if(!useValueMap_ && itPF->pt() == 0) pVec.SetPxPyPzE(phoCands[iPho]->px()*pWeight,phoCands[iPho]->py()*pWeight,phoCands[iPho]->pz()*pWeight,phoCands[iPho]->energy()*pWeight);
>
> I have a sense of deja vu.
>
> This is our Mar PR
> violatingcp@1f1c606#diff-56c51df824006a6f9021c77e89eee895
> <https://github.com/violatingcp/cmssw/commit/1f1c606c4640b33f7274b996ad66d6821add8b10#diff-56c51df824006a6f9021c77e89eee895>
> that removed this line
> and it was removed to fix a problem with the MET tail
> https://cloud.githubusercontent.com/assets/4676718/13655024/39dc7c98-
> e611-11e5-99db-86497785624c.png
> #13115 (comment)
> <https://github.com/cms-sw/cmssw/pull/13115#issuecomment-194585674>
>
> If I understand correctly, what we see here is that some of the photon's
> associated cands (associatedPackedPFCandidates) are puppi pileups, probably
> on top of a good non-pu puppi cand. For all these pileup puppi cands you
> are generating the full photon-sized MET candidate.
> If this line was added for a good reason, there should probably be an
> accumulation of all other associated cands so that the photon's momentum is
> not exceeded.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15562/files/691379a0f452cf21724f83c3978bf13682245d8b#r77749405>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AE7V4ZARFnI7rSJO6BN-xqnjpCN5QTWcks5qnhlagaJpZM4JqyuB>
> .
>